owasp-top10-2017-apps/a5/ecommerce-api/app/handlers/handlers.go

```diff
@@ -13,14 +14,40 @@ func HealthCheck(c echo.Context) error {
 	return c.String(http.StatusOK, "WORKING\n")
 }
 
+func ErrorMessage(c echo.Context) error {
+	return c.JSON(http.StatusBadRequest, map[string]string{"result": "error", "details": "Error finding this UserID."})
```

Comment:
Nice! It is very important to avoid giving specific internals information to the user when an error occurs. 

In this particular what do you think of using `"Error: Permission Denied. You are not allowed to view this ticket."`, or something similar instead?

Also, using a 403 status code (`http.StatusForbidden`) would be a good idea! 😃